src/parser/ts/grammar.ne

```diff
@@ -108,12 +123,12 @@ function -> unary br arithmethicElements BR {% extractFunction %}
 oneElement -> element {% extractCommaSeparatedArray %}
 twoElements -> compositeElement comma compositeElement {% extract2Elements %}
 commaSeparatedElements -> compositeElement (comma compositeElement):* {% extractCommaSeparatedArray %}
-arithmethicElements -> element (operator element):+ {% extractOperationsArray %}
+arithmeticElements -> (unary_operator):? element (operator (unary_operator):? element):+ {% extractOperationsArray %}
 
 map_func_args -> compositeElement "," array "," array comma number {% extractMapFuncArgs %}
 
-compositeElement -> element {% id %}
-  | arithmethicElements {% id %}
+compositeElement -> (unary_operator):? element {% extractUnaryOperator %}
+  | arithmeticElements {% id %}
```

Comment:
it seems like a bit too much of grammar / logic for this minor feature.
can we try to addd it just in one place?

like in `function` for instance, since it's used as an "element" everywhere?

Reply:
it will produce additional ambiguity if I'll add this logic to `function` instead